### ISSUE TITLE
Add some integration tests to detect bad network behaviour

### ISSUE THREAD

**Issue by Chicoryn** [Chicoryn/dream-go#15]
Add an integration test that checks what policy and value is predicted for some board positions with the following features:

* Big dead dragon (with one eye), check that it scores the game correctly.
* Broken ladders, check that the policy does not suggest playing it out.

Some suggested SGF files containing either broken ladders (both from leela-zero, but we should check them anyway) or big dragons that are dead.

[big_dead_dragon.sgf.gz](https://github.com/Chicoryn/dream-go/files/1650318/big_dead_dragon.sgf.gz)
[ladder_1.sgf.gz](https://github.com/Chicoryn/dream-go/files/1650336/ladder_1.sgf.gz)
[ladder_2.sgf.gz](https://github.com/Chicoryn/dream-go/files/1650337/ladder_2.sgf.gz)

**Comment by Chicoryn** [comment]
Might be of interest:

https://github.com/cmaughan/cgtc

**Comment by Chicoryn** [comment]
I just added the following game, based on one I played between Leela 0.11.0 and some internal weights. In the actual game the network misjudged the final board position significantly because of the random black stones that are placed in whites territory, as soon as they were removed the situation improved:

[end_1.sgf.gz](https://github.com/Chicoryn/dream-go/files/1687680/end_1.sgf.gz)

**Comment by Chicoryn** [comment]
Added one more ladder test from a leela-zero game that was posted on cbaduk:

[ladder_3.sgf.gz](https://github.com/Chicoryn/dream-go/files/1708361/ladder_3.sgf.gz)

**Comment by Chicoryn** [comment]
Added one more game with some large dead dragons that doesn't register. This seems to be a consistent problem where it doesn't detect that a big dragon is dead in the value network and it affects everything when it comes to value of play.

Part of the reason for this might be the reinforcement learning algorithm that we are using, as it does not generate very high quality games for the value network. It might also be the 0.01 coefficient for the MSE component of the loss function.

[dead_dragon_2.sgf.gz](https://github.com/Chicoryn/dream-go/files/1712145/dead_dragon_2.sgf.gz)

**Comment by Chicoryn** [comment]
I tweaked the regularization coefficient a little bit (which is equivalent to increasing all of the other coefficients) and we got a network with (very) slightly better value loss (~0.02). However it still has issues identifying large dead groups:

[dead_dragon_3.sgf.gz](https://github.com/Chicoryn/dream-go/files/1714384/dead_dragon_3.sgf.gz)

---

After some investigation into the linked game one can make the following observation:

- The dead dragon is considered alive by the scoring engine. This means we teach the neural network that said dragon is alive, which could explain the issues it is having with determining that the group is dead.

**Comment by Chicoryn** [comment]
I added a few more that I've had laying around from when I revamped the scoring engine. They mostly deal with _seki_, _dead dragons_, or _bent four in the corner_.

In general the neural network fails them horribly, but with the recent advances using dilation we might start seeing less failure soon™ (I hope, since preliminary results were not too promising).

[bent_four_1.sgf.gz](https://github.com/Chicoryn/dream-go/files/1866629/bent_four_1.sgf.gz)
[dead_dragon_4.sgf.gz](https://github.com/Chicoryn/dream-go/files/1866630/dead_dragon_4.sgf.gz)
[seki_1.sgf.gz](https://github.com/Chicoryn/dream-go/files/1866631/seki_1.sgf.gz)
[seki_2.sgf.gz](https://github.com/Chicoryn/dream-go/files/1866632/seki_2.sgf.gz)
[seki_3.sgf.gz](https://github.com/Chicoryn/dream-go/files/1866633/seki_3.sgf.gz)
[seki_4.sgf.gz](https://github.com/Chicoryn/dream-go/files/1866634/seki_4.sgf.gz)
